moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1163,6 +1178,11 @@ bool RobotState::getJacobian(const JointModelGroup* group, const LinkModel* link
     const JointModel* pjm = link->getParentJointModel();
     if (pjm->getVariableCount() > 0)
     {
+      if (not group->hasJointModel(pjm->getName()))
```

Comment:
Wouldn't it suffice to continue here with `link = pjm->getParentLinkModel()` to skip all joints that are not part of group (and thus assume they are fixed)?

Reply:
That is a better idea, I'll make that change soon.